tests/terraform/modules/define_node_role.sh

```diff
@@ -20,51 +19,27 @@ cp_worker_nodes=$8
 # Set the desired role into an array based on the index
 order_array=($(echo "$role_order" | tr ',' '\n'))
 role_array=()
-for order_num in "${order_array[@]}"
-do
-  if [[ "$order_num" = "1" ]]
-  then
-    until [ "$all_role_nodes" -le "0" ]
-    do
-       role_array+=("all-roles")
-       ((all_role_nodes-=1))
-    done
-  elif [[ "$order_num" = "2" ]]
-  then
-    until [ "$etcd_only_nodes" -le "0" ]
-    do
-       role_array+=("etcd-only")
-       (("etcd_only_nodes-=1"))
-    done
-  elif [[ "$order_num" = "3" ]]
-  then
-    until [ "$etcd_cp_nodes" -le "0" ]
-    do
-       role_array+=("etcd-cp")
-       (("etcd_cp_nodes-=1"))
-    done
-  elif [[ "$order_num" = "4" ]]
-  then
-    until [ "$etcd_worker_nodes" -le "0" ]
-    do
-       role_array+=("etcd-worker")
-       (("etcd_worker_nodes-=1"))
-    done
-  elif [[ "$order_num" = "5" ]]
-  then
-    until [ "$cp_only_nodes" -le "0" ]
-    do
-       role_array+=("cp-only")
-       (("cp_only_nodes-=1"))
-    done
-  elif [[ "$order_num" = "6" ]]
-  then
-    until [ "$cp_worker_nodes" -le "0" ]
-    do
-       role_array+=("cp-worker")
-       (("cp_worker_nodes-=1"))
-    done
-  fi
+for order_num in "${order_array[@]}"; do
+  case "$order_num" in
+    1)
+      role_array+=($(printf "all-roles %.0s " $(seq 1 "$all_role_nodes")))
+      ;;
+    2)
+      role_array+=($(printf "etcd-only %.0s " $(seq 1 "$etcd_only_nodes")))
+      ;;
+    3)
+      role_array+=($(printf "etcd-cp %.0s " $(seq 1 "$etcd_cp_nodes")))
+      ;;
+    4)
+      role_array+=($(printf "etcd-worker %.0s " $(seq 1 "$etcd_worker_nodes")))
+      ;;
+    5)
+      role_array+=($(printf "cp-only %.0s " $(seq 1 "$cp_only_nodes")))
+      ;;
+    6)
+      role_array+=($(printf "cp-worker %.0s " $(seq 1 "$cp_worker_nodes")))
+      ;;
+  esac
```

Comment:
I hate bash scripting 😂 

To me, this is less readable even though it's less lines of code and looks neater. I'm fine with it though if this is what the team wants to have and if it's still functionally the same.

Reply:
i am totally ok with any, i just changed bc i had a really hard time trying to understand this script , specially this part,  for me is much clear and readable this way .
but again lets wait for what people think.  😂